### DISCUSSION TITLE
POST /api/v1/accounts 405 Method Not Allowed

### DISCUSSION THREAD

**Discussion by xEtarusx** [firefly-iii/firefly-iii#6722] [Firefly III] [answered]
Hello,
i'm trying to add asset accounts via the API /api/v1/accounts endpoint (https://api-docs.firefly-iii.org/#/accounts/storeAccount) but I get a 405 Method Not Allowed as response.

I'm using POST with the Authorization Bearer Token and my personal access token created in Firefly III and the raw body
`{"name":"Test Account", "type":"asset"}` as the API documentation say that these two are the only required fields.

My Firefly III instance is running in a docker container behind Traefik.

Docker logs:
```
fireflyiii     | 172.18.0.1 - - [18/Dec/2022:11:17:05 +0100] "POST /api/v1/accounts HTTP/1.1" 302 1324 "-" "PostmanRuntime/7.30.0"
fireflyiii     | 172.18.0.1 - - [18/Dec/2022:11:17:05 +0100] "POST / HTTP/1.1" 405 1767 "http://firefly.example.com/api/v1/accounts" "PostmanRuntime/7.30.0"
```

Does anybody know why I get the 405?

Thank you
Etarus

**Comment by JC5** [comment, marked as answer]
>  docker container behind Traefik.

Could it be that Traefik is blocking this?

The post you need to send must include the `account_role` field when the `type` is asset:

```
{
"name": "test1",
"type": "asset",
"account_role": "defaultAsset"
}
```

**Reply by xEtarusx** [reply]
The account_role was missing. Now it works, thank you.
I might have overlooked that this was required too in the API documentation.

**Reply by JC5** [reply]
Glad to help. However, it should not return a 405 when it's missing information, that may be something to look at.
